src/NHibernate/Dialect/Schema/PostgreSQLMetadata.cs

```diff
@@ -138,7 +138,43 @@ public PostgreSQLColumnMetadata(DataRow rs)
 			this.SetNumericalPrecision(rs["NUMERIC_PRECISION"]);
 
 			Nullable = Convert.ToString(rs["IS_NULLABLE"]);
-			TypeName = Convert.ToString(rs["DATA_TYPE"]);
+			TypeName = NormalizeTypeNames(Convert.ToString(rs["DATA_TYPE"]));
+		}
+
+		private static string NormalizeTypeNames(string typeName)
+		{
+			switch (typeName)
+			{
+				case "double precision":
+					return "float8";
+				case "real":
+					return "float4";
+				case "smallint":
+					return "int2";
+				case "integer":
+					return "int4";
+				case "bigint":
+					return "int8";
+			}
+
+			if (typeName.StartsWith("character", StringComparison.Ordinal))
+			{
+				return typeName.Replace("character varying", "varchar").Replace("character", "char");
```

Comment:
It is likely we are fixing another old trouble by the way here: for fixed length character types, the older Npgsql drivers could yield `bpchar`, while our dialects are using `char`.